Mathlib/Condensed/Discrete/LocallyConstant.lean

```diff
@@ -0,0 +1,499 @@
+    (X.val.mapIso (opCoproductIsoProduct' (finiteCoproduct.isColimit.{u, u} fun a ↦ of (σ a))
+      (productIsProduct fun x ↦ Opposite.op (of (σ x))))).hom ≫
+    (PreservesProduct.iso X.val fun a ↦ ⟨of (σ a)⟩).hom ≫
+    (Types.productIso.{u, u + 1} fun a ↦ X.val.obj ⟨of (σ a)⟩).hom := by
+  ext x a
+  simp only [finiteCoproduct.cocone_pt, Fan.mk_pt, Functor.mapIso_hom,
+    PreservesProduct.iso_hom, types_comp_apply, Types.productIso_hom_comp_eval_apply]
+  have := congrFun (piComparison_comp_π X.val (fun a ↦ ⟨of (σ a)⟩) a)
+  simp only [types_comp_apply] at this
+  rw [this, ← FunctorToTypes.map_comp_apply]
+  simp only [sigmaComparison]
+  apply congrFun
+  congr 2
+  erw [← opCoproductIsoProduct_inv_comp_ι]
+  simp only [coe_of, Opposite.unop_op, unop_comp, Quiver.Hom.unop_op, Category.assoc]
+  change finiteCoproduct.ι.{u, u} (fun a ↦ of (σ a)) _ = _
+  rw [← Sigma.ι_comp_toFiniteCoproduct]
+  congr
+  simp only [opCoproductIsoProduct, ← unop_comp, coproductIsoCoproduct,
+    opCoproductIsoProduct'_comp_self]
+  rfl
+
+instance : IsIso <| sigmaComparison X σ := by
+  rw [sigmaComparison_eq_comp_isos]
+  infer_instance
+
+end SigmaComparison
+
+namespace LocallyConstant
+
+/--
+The functor from the category of sets to presheaves on `CompHaus` given by locally constant maps.
+-/
+@[simps]
+def functorToPresheaves : Type* ⥤ (CompHaus.{u}ᵒᵖ ⥤ Type _) where
+  obj X := {
+    obj := fun ⟨S⟩ ↦ LocallyConstant S X
+    map := fun f g ↦ g.comap f.unop }
+  map f := { app := fun S t ↦ t.map f }
+
+/--
+Locally constant maps are the same as continuous maps when the target is equipped with the discrete
+topology
+-/
+@[simps]
+def locallyConstantIsoContinuousMap (Y X : Type*) [TopologicalSpace Y] :
+    LocallyConstant Y X ≅ C(Y, TopCat.discrete.obj X) :=
+  letI : TopologicalSpace X := ⊥
+  haveI : DiscreteTopology X := ⟨rfl⟩
+  { hom := fun f ↦ (f : C(Y, X))
+    inv := fun f ↦ ⟨f, (IsLocallyConstant.iff_continuous f).mpr f.2⟩ }
+
+/-- `locallyConstantIsoContinuousMap` is a natural isomorphism. -/
+noncomputable def functorToPresheavesIsoTopCatToCondensed (X : Type (u+1)) :
+    functorToPresheaves.obj X ≅ (topCatToCondensed.obj (TopCat.discrete.obj X)).val :=
+  NatIso.ofComponents (fun S ↦ locallyConstantIsoContinuousMap _ _)
+
+/-- `Condensed.LocallyConstant.functorToPresheaves` lands in condensed sets. -/
+@[simps]
+def functor : Type (u+1) ⥤ CondensedSet.{u} where
+  obj X := {
+    val := functorToPresheaves.obj X
+    cond := by
+      rw [Presheaf.isSheaf_of_iso_iff (functorToPresheavesIsoTopCatToCondensed X)]
+      exact (topCatToCondensed.obj _).cond
+  }
+  map f := ⟨functorToPresheaves.map f⟩
+
+/--
+`Condensed.LocallyConstant.functor` is naturally isomorphic to the restriction of
+`topCatToCondensed` to discrete topological spaces.
+-/
+noncomputable def functorIsoTopCatToCondensed : functor ≅ TopCat.discrete ⋙ topCatToCondensed :=
+  NatIso.ofComponents (fun X ↦ {
+    hom := ⟨(functorToPresheavesIsoTopCatToCondensed X).hom⟩
```

Comment:
It should be possible to use `Functor.preimageIso` here (applied to the forget functor from sheaves to presheaves).

Reply:
That works, thanks!